__tests__/unit/fetch-client.test.ts

```diff
@@ -0,0 +1,113 @@
+import { enableFetchMocks } from "jest-fetch-mock";
+enableFetchMocks();
+
+import {
+  FetchClient,
+  HTTPRequest,
+  HTTPResponse,
+  isHTTPResponse,
+} from "../../src/http-client";
+import { QueryFailure, QuerySuccess } from "../../src/wire-protocol";
+
+let fetchClient: FetchClient;
+
+const dummyRequest: HTTPRequest = {
+  data: { query: "" },
+  headers: {},
+  method: "POST",
+  url: "",
+};
+
+const dummyStats = {
+  compute_ops: 0,
+  read_ops: 0,
+  write_ops: 0,
+  query_time_ms: 0,
+  storage_bytes_read: 0,
+  storage_bytes_written: 0,
+  contention_retries: 0,
+};
+
+describe("query", () => {
+  beforeAll(() => {
+    fetchClient = new FetchClient();
+  });
+
+  beforeEach(() => {
+    fetchMock.resetMocks();
+  });
+
+  it("returns a valid query response on success", async () => {
+    expect.assertions(7);
+    const expected: QuerySuccess<null> = {
+      data: null,
+      summary: "summary",
+      txn_time: "2023-02-17T19:19:12.887330Z",
+      query_tags: {},
```

Comment:
minor nit

- let's make `query_tags` and `data` have some value.

That way the test verifies we're doing the work on the following lines:

https://github.com/fauna/fauna-js/blob/main/src/http-client/fetch-client.ts#L28

https://github.com/fauna/fauna-js/blob/main/src/http-client/fetch-client.ts#L30

Reply:
sounds good. If you want both go for it.